agent/handlers/v4/tmdsstate.go

```diff
@@ -149,3 +153,49 @@ func (s *TMDSAgentState) getTaskMetadata(v3EndpointID string, includeTags bool)
 
 	return *taskResponse, nil
 }
+func (s *TMDSAgentState) GetContainerStats(v3EndpointID string) (tmdsv4.StatsResponse, error) {
+	taskARN, ok := s.state.TaskARNByV3EndpointID(v3EndpointID)
+	if !ok {
+		return tmdsv4.StatsResponse{}, tmdsv4.NewErrorStatsLookupFailure(fmt.Sprintf(
+			"V4 container handler: unable to get task arn from request: unable to get task Arn from v3 endpoint ID: %s",
+			v3EndpointID))
+	}
+
+	containerID, ok := s.state.DockerIDByV3EndpointID(v3EndpointID)
+	if !ok {
+		return tmdsv4.StatsResponse{}, tmdsv4.NewErrorStatsLookupFailure(fmt.Sprintf(
+			"V4 container stats handler: unable to get container ID from request: unable to get docker ID from v3 endpoint ID: %s",
+			v3EndpointID))
+	}
+
+	dockerStats, network_rate_stats, err := s.statsEngine.ContainerDockerStats(taskARN, containerID)
+	if err != nil {
+		return tmdsv4.StatsResponse{}, tmdsv4.NewErrorStatsFetchFailure(fmt.Sprintf(
```

Comment:
should be put the actual `err` in the input for `NewErrorStatsFetchFailure` so that its not lost in translation?

Reply:
Yeah that's a good catch. Let me test that.

Reply:
I have now added a `cause` field to this error type.